saleor/dashboard/emails.py

```diff
@@ -11,35 +13,69 @@
 from ..core.utils import build_absolute_uri
 
 
-def _send_set_password_email(pk, template_name):
-    user = User.objects.get(pk=pk)
-    uid = urlsafe_base64_encode(force_bytes(user.pk))
-    token = default_token_generator.make_token(user)
+def send_set_password_staff_email_with_url(redirect_url, staff):
+    """Trigger sending a set password email for the given staff."""
+    token = default_token_generator.make_token(staff)
+    _send_set_user_password_email_with_url.delay(
+        staff.email, redirect_url, token, "dashboard/staff/set_password"
+    )
+
+
+def send_set_password_customer_email_with_url(redirect_url, customer):
+    """Trigger sending a set password email for the given customer."""
+    token = default_token_generator.make_token(customer)
+    _send_set_user_password_email_with_url.delay(
+        customer.email, redirect_url, token, "dashboard/customer/set_password"
+    )
+
+
+def send_set_password_staff_email(staff):
+    """Trigger sending a set password email for the given staff."""
+    token = default_token_generator.make_token(staff)
+    _send_set_user_password_email.delay(
+        staff.email, staff.pk, token, "dashboard/staff/set_password"
+    )
+
+
+def send_set_password_customer_email(customer):
+    """Trigger sending a set password email for the given customer."""
+    token = default_token_generator.make_token(customer)
+    _send_set_user_password_email.delay(
+        customer.email, customer.pk, token, "dashboard/customer/set_password"
+    )
+
+
+@app.task
+def _send_set_user_password_email_with_url(
+    recipient_email, redirect_url, token, template_name
+):
+    params = urlencode({"email": recipient_email, "token": token})
+    password_set_url = f"{redirect_url}?%{params}"
```

Comment:
I think `urllib.parse` has some tools for that.

Reply:
You would have to split the url then unsplit it